azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageTypeDaoImpl.java

```diff
@@ -122,6 +122,32 @@ public int createImageType(final ImageType imageType) {
     return imageTypeId;
   }
 
+  @Override
+  public ImageType getImageTypeWithOwnershipsById(final String id) {
+    final FetchImageTypeHandler fetchImageTypeHandler = new FetchImageTypeHandler();
+    List<ImageType> imageTypes = new ArrayList<>();
+    try {
+      imageTypes = this.databaseOperator.query(FetchImageTypeHandler.FETCH_IMAGE_TYPE_BY_ID,
+          fetchImageTypeHandler, id);
+      if (imageTypes == null || imageTypes.isEmpty()) {
```

Comment:
Is the null check needed? I think we should guarantee a non null result.

Reply:
Again, yes and no.

Reply:
Fixed.